### DISCUSSION TITLE
Typing the event handler context

### DISCUSSION THREAD

**Discussion by targetlucked69** [nuxt/nuxt#17543] [General]
Is it possible to type the context in event handler?

```ts
export default defineEventHandler((event) => {
  event.req.context.someProperty // string
})
```

**Comment by viii47** [comment]
You can define your own interface with the properties that you expect and declare the context with it.

```ts
interface IAmAwesome {
    yes?: string;
    no?: string;
    maybe?: string;
}

export default defineEventHandler(async (event) => {
    const context: IAmAwesome = event.req.context;
    context...
})
```

![image](https://user-images.githubusercontent.com/95022556/170005946-998abf59-df31-4cd2-8632-36002701592a.png)

**Comment by pi0** [comment]
Hi! Here is the related discussion you can follow: https://github.com/unjs/h3/issues/122
